Content.Server/Radio/Components/HeadsetComponent.cs

```diff
@@ -1,27 +1,57 @@
+using Robust.Shared.Audio;
+using Robust.Shared.Containers;
 using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
 
 namespace Content.Server.Radio.Components;
-
 /// <summary>
 ///     This component relays radio messages to the parent entity's chat when equipped.
 /// </summary>
 [RegisterComponent]
 [Access(typeof(HeadsetSystem))]
 public sealed class HeadsetComponent : Component
 {
-    [DataField("channels", customTypeSerializer: typeof(PrototypeIdHashSetSerializer<RadioChannelPrototype>))]
-    public readonly HashSet<string> Channels = new() { "Common" };
+    /// <summary>
+    ///     This variable indicates locked state of encryption keys, allowing or prohibbiting insert/remove them from headset.
+    ///     true  => Human will able to screw out encryption keys with tool mentioned in KeysExtractionMethod and will be able to put encryption keys in headset.
+    ///     false => encryption keys will be locked in headset, there will be no proper way to extract them or to add keys anymore.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("isKeysLocked")]
+    public bool IsKeysLocked = true;
+    /// <summary>
+    ///     Shows what tool human should use to extract encryption keys from headset.
+    ///     default "Screwing" means it is extracting by screwdriver.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("keysExtractionMethod", customTypeSerializer: typeof(PrototypeIdSerializer<ToolQualityPrototype>))]
+    public string KeysExtractionMethod = "Screwing";
 
-    // Maybe make the defaultChannel an actual channel type some day, and use that for parsing messages
-    // [DataField("defaultChannel", customTypeSerializer: typeof(PrototypeIdHashSetSerializer<RadioChannelPrototype>))]
-    // public readonly HashSet<string> defaultChannel = new();
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("keySlots")]
+    public int KeySlots = 2;
+
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("keyExtractionSound")]
+    public SoundSpecifier KeyExtractionSound = new SoundPathSpecifier("/Audio/Items/pistol_magout.ogg");
 
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("keyInsertionSound")]
+    public SoundSpecifier KeyInsertionSound = new SoundPathSpecifier("/Audio/Items/pistol_magin.ogg");
 
+    [ViewVariables]
+    public Container KeyContainer = default!;
+    public const string KeyContainerName = "key_slots";
 
+    [ViewVariables]
+    public HashSet<string> Channels = new();
+
+    // Maybe make the defaultChannel an actual channel type some day, and use that for parsing messages
+    // [DataField("defaultChannel", customTypeSerializer: typeof(PrototypeIdHashSetSerializer<RadioChannelPrototype>))]
+    // public readonly HashSet<string> defaultChannel = new();
     [DataField("defaultChannel", customTypeSerializer: typeof(PrototypeIdSerializer<RadioChannelPrototype>))]
     public readonly string? defaultChannel;
```

Comment:
Have you considered how using the :h prefix is going to work with encryption keys? Currently defaultChannel is on the still headset prototype and that will cause headsets to try and use a channel they don't have access to anymore if you replace their key.

Reply:
Maybe, what if put defaultChannel in first-inserted-in-headset key?

Reply:
It's okay, if headset will not have DefaultChannel in Channels it is predicted already